[25.0] Improve dataset collection fetch performance in the invocation view

ahmedhamidawan opened the pull request:
Fixes the 25.0 part of https://github.com/galaxyproject/galaxy/issues/20864. 

- We lazy load the input/output tabs as well
- We fetch "collection" view for HDCAs in `GenericItem` and `useInvocationGraph` by default
  - For the generic content items (used outside of the history panel), there is no need to fetch the entire "element" view when displaying the content item unexpanded. Therefore, I have made the default view "collection", until you click to expand an HDCA which is when the `DatasetCollectionProvider` is remounted to fetch the "element" (detailed) view for the HDCA.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

ahmedhamidawan committed:
[25.0] Make invocation input/output tabs lazy

Addresses https://github.com/galaxyproject/galaxy/issues/20864

ahmedhamidawan committed:
fetch "collection" view for HDCAs in `GenericItem` and `useInvocationGraph`

For the generic content items (used outside of the history panel), there is no need to fetch the entire "element" view when displaying the content item unexpanded. Therefore, I have made the default view "collection", until you click to expand an HDCA which is when the `DatasetCollectionProvider` is remounted to fetch the "element" (detailed) view for the HDCA.

Co-authored-by: Marius van den Beek <m.vandenbeek@gmail.com>

mvdbeek commented on client/src/api/datasetCollections.ts:
```diff
@@ -18,9 +18,12 @@ export type CollectionType = string;
+export async function fetchCollectionDetails(
+    params: { hdca_id: string },
+    query?: { view?: string }
+): Promise<HDCADetailed> {
```
Is this the right return type ? It should be `HDCASummary` for `view="collection"` i think. I would create a second function here (`fetchCollectionSummary`) with a singular return type or make this generic on the view param.

ahmedhamidawan replied:
Good catch!

ahmedhamidawan replied:
I totally missed the `fetchCollectionSummary` function right below it 🤦

mvdbeek replied:
I did not see it either, but hooray for consistent naming 😄

ahmedhamidawan committed:
use a `tabsNotLazy` prop to make input/output tabs jest work

This seems to be the simplest workaround since we do not have access to the actual, clickable tab links in this component (to be able to trigger the tab display).
